web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -390,6 +397,82 @@ $.extend(fixmystreet.set_up, {
     });
   },
 
+  list_item_actions: function() {
+    $('.item-list').on('click', ':submit', function(e) {
+      e.preventDefault();
+
+      var $submitButton = $(this);
+      var whatUserWants = $submitButton.prop('name');
+      var data;
+      var $item;
+      var $list;
+      var $hiddenInput;
+      if (fixmystreet.page === 'around') {
+          // Deal differently because one big form
+          var parts = whatUserWants.split('-');
+          whatUserWants = parts[0] + '-' + parts[1];
+          var id = parts[2];
+          var token = $('[name=token]').val();
+          data = whatUserWants + '=1&token=' + token + '&id=' + id;
+      } else {
+          var $form = $(this).parents('form');
+          $item = $form.parent('.item-list__item');
+          $list = $item.parent('.item-list');
+
+          // The server expects to be told which button/input triggered the form
+          // submission. But $form.serialize() doesn't know that. So we inject a
+          // hidden input into the form, that can pass the name and value of the
+          // submit button to the server, as it expects.
+          $hiddenInput = $('<input>').attr({
+            type: 'hidden',
+            name: whatUserWants,
+            value: $submitButton.prop('value')
+          }).appendTo($form);
+          data = $form.serialize() + '&ajax=1';
+      }
+
+      // Update UI while the ajax request is sent in the background.
+      if ('shortlist-down' === whatUserWants) {
+        $item.insertAfter( $item.next() );
+      } else if ('shortlist-up' === whatUserWants) {
+        $item.insertBefore( $item.prev() );
+      } else if ('shortlist-remove' === whatUserWants) {
+          $submitButton.attr('class', 'item-list__item__shortlist-add');
+          $submitButton.attr('name', 'shortlist-add');
+          // TODO Change title/value to Add
+      } else if ('shortlist-take' === whatUserWants) {
+          $submitButton.attr('class', 'item-list__item__shortlist-remove');
+          $submitButton.attr('name', 'shortlist-remove');
+          // TODO Change title/value to Remove
+      } else if ('shortlist-add' === whatUserWants) {
```

Comment:
shortlist-take doesn’t exist as a button any more, it’s only for a different icon, so yes, one can be removed :)